python/pyarrow/_compute.pyx

```diff
@@ -1131,106 +1092,83 @@ cdef class _SplitPatternOptions(FunctionOptions):
+        return CSortOrder_Ascending
+    elif order == "descending":
+        return CSortOrder_Descending
+    else:
+        raise ValueError(f'"{order}" is not a valid order')
+
+
 cdef class _ArraySortOptions(FunctionOptions):
     def _set_options(self, order):
-        if order == "ascending":
-            self.wrapped.reset(new CArraySortOptions(CSortOrder_Ascending))
-        elif order == "descending":
-            self.wrapped.reset(new CArraySortOptions(CSortOrder_Descending))
-        else:
-            raise ValueError(
-                "{!r} is not a valid order".format(order)
-            )
+        self.wrapped.reset(new CArraySortOptions(unwrap_sort_order(order)))
 
 
 class ArraySortOptions(_ArraySortOptions):
-    def __init__(self, *, order='ascending'):
+    def __init__(self, *, order="ascending"):
         self._set_options(order)
 
 
 cdef class _SortOptions(FunctionOptions):
     def _set_options(self, sort_keys):
-        cdef:
-            vector[CSortKey] c_sort_keys
-            c_string c_name
-            CSortOrder c_order
-
+        cdef vector[CSortKey] c_sort_keys
         for name, order in sort_keys:
-            if order == "ascending":
-                c_order = CSortOrder_Ascending
-            elif order == "descending":
-                c_order = CSortOrder_Descending
-            else:
-                raise ValueError(
-                    "{!r} is not a valid order".format(order)
-                )
-            c_name = tobytes(name)
-            c_sort_keys.push_back(CSortKey(c_name, c_order))
-
+            c_sort_keys.push_back(CSortKey(tobytes(name),
+                                  unwrap_sort_order(order)))
         self.wrapped.reset(new CSortOptions(c_sort_keys))
 
 
 class SortOptions(_SortOptions):
-    def __init__(self, sort_keys=None):
+    def __init__(self, *, sort_keys=None):
         if sort_keys is None:
             sort_keys = []
         self._set_options(sort_keys)
 
 
 cdef class _SelectKOptions(FunctionOptions):
     def _set_options(self, k, sort_keys):
-        cdef:
-            c_string c_name
-            vector[CSortKey] c_sort_keys
-            CSortOrder c_order
-
+        cdef vector[CSortKey] c_sort_keys
         for name, order in sort_keys:
-            if order == "ascending":
-                c_order = CSortOrder_Ascending
-            elif order == "descending":
-                c_order = CSortOrder_Descending
-            else:
-                raise ValueError(
-                    "{!r} is not a valid order".format(order)
-                )
-            c_name = tobytes(name)
-            c_sort_keys.push_back(CSortKey(c_name, c_order))
-
+            c_sort_keys.push_back(CSortKey(tobytes(name),
+                                  unwrap_sort_order(order)))
         self.wrapped.reset(new CSelectKOptions(k, c_sort_keys))
 
 
 class SelectKOptions(_SelectKOptions):
-    def __init__(self, k, sort_keys=None):
+    def __init__(self, *, k=-1, sort_keys=None):
```

Comment:
Well, it should be clear that the first argument to `select_k` is the `k` number, no?